### ISSUE TITLE
Normal AI worker stuck at field ends

### ISSUE THREAD

**Issue by pvaiko** [Courseplay/Courseplay_FS22#2337]
### Discussed in https://github.com/Courseplay/Courseplay_FS22/discussions/2332

<div type='discussions-op-text'>

<sup>Originally posted by **danitbit** February 16, 2023</sup>
When I have the courseplay mod enabled, if I run a normal AI worker on a field and I have that vehicle focused, they will not turn at the end of a field. If I exit the vehicle before they get to the end of the field, the AI turns just fine. If I switch to the vehicle as they are turning, they will not continue to work the field. They get stuck. Is this something that you would like a more detailed bug submission for?</div>

**Comment by Azthek** [comment]
Hello, I have this issue as well.  I stream through GeforceNow, so I dont normally have access to savegame or logs, but I got to test this on a different pc (where I can run the game natively).
Same behaviour as described in op, however it seems to affect only trailed equipment - small 3 point equipment such as starting seeder or cultivator are not affected (at least from my limited testing).
Tested on a brand new elm creek savegame, only courseplay enabled (on GFN I only have access to ingame modhub so 7.2.0.5, but on the on the other pc I tested latest github release - 7.2.1.3), 
Default Giants helper gets stuck at the end of the field not able to start the turn manouver - but only if the vehicle is focused at the time of transition from the end of the lane to start of turn manouver. If Im outside of the vehicle regardless of the distance, there is no issue. 
Helper can get stuck at the start of the field as well, if the equipment is not perfectly aligned right on the edge of the field.
Disabling courseplay fixes the issues and helper operates normally.
Im attaching 2 logs which captured the issue, one with a roller, and other with disc harrow (both trailed).
[log.txt](https://github.com/Courseplay/Courseplay_FS22/files/11385129/log.txt)
[log-1.txt](https://github.com/Courseplay/Courseplay_FS22/files/11385131/log-1.txt)

I understand this might not be entirely courseplay at fault, might even be Giants bug, but nowhere else to get help with modded games. Also this is not a gamebreaking bug, its just annoing. 
Anyway, any help is appreciated. Thanks.
